### PR TITLE
[HGCAL TPG] Updates since 10_5_0_pre1

### PR DESCRIPTION
**Pull request by jbsauvan**
Add the following updates on top of 10_5_0_pre1:
- PFCal-dev/cmssw#259
- PFCal-dev/cmssw#262

### CODE REVIEW

**Comment by rekovic** [comment]
@jbsauvan
This causes PF Calo Producer not to compile:

```
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_5_0_pre1-Phase2-v3/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFCaloProducer.cc: In member function 'void L1TPFCaloProducer::readHcalHGCTCs_(edm::Event&, const edm::EventSetup&)':
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_5_0_pre1-Phase2-v3/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFCaloProducer.cc:251:13: error: 'HGCalDetId' was not declared in this scope
             HGCalDetId id(it->detId());
             ^~~~~~~~~~
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_5_0_pre1-Phase2-v3/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFCaloProducer.cc:251:13: note: suggested alternative: 'HcalDetId'
             HGCalDetId id(it->detId());
             ^~~~~~~~~~
             HcalDetId
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_5_0_pre1-Phase2-v3/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFCaloProducer.cc:252:23: error: 'id' was not declared in this scope
             bool em = id.subdetId() == HGCEE;
                       ^~
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_5_0_pre1-Phase2-v3/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFCaloProducer.cc:252:23: note: suggested alternative: 'ed'
             bool em = id.subdetId() == HGCEE;
                       ^~
                       ed
/afs/cern.ch/work/r/rekovic/private/relaeses/10x/CMSSW_10_5_0_pre1-Phase2-v3/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1TPFCaloProducer.cc:252:40: error: 'HGCEE' was not declared in this scope
             bool em = id.subdetId() == HGCEE;
```

Taking a look.

File: DataFormats/L1THGCal/interface/HGCalClusterT.h [hunk 1]
```diff
@@ -148,8 +150,6 @@ namespace l1t
       }
 
       uint32_t subdetId() const {return detId_.subdetId();} 
-      uint32_t layer() const {return detId_.layer();}
```
Comment (rekovic) [inline DataFormats/L1THGCal/interface/HGCalClusterT.h:151]: @jbsauvan 
What happens to this accessor and the line below ?

**Comment by gpetruc** [comment]
@rekovic @jbsauvan 

I think we could remove the `L1TPFCaloProducer::readHcalHGCTCs_` function as it's not needed anymore in the current workflow.
I could give you a PR on top of this PR tomorrow.

Giovanni

File: DataFormats/L1THGCal/interface/HGCalClusterT.h (hunk 1 above)
Comment (jbsauvan) [reply DataFormats/L1THGCal/interface/HGCalClusterT.h:151]: `detId_` needs to be generic (`DetId`) in order to be able to deal with the different detid schemes. The `layer` should now be retrieved through the `HGCalTriggerTools`:
https://github.com/PFCal-dev/cmssw/blob/hgc-tpg-devel-CMSSW_10_5_0_pre1/L1Trigger/L1THGCal/interface/HGCalTriggerTools.h#L46

**Comment by rekovic** [comment]
Fix of removeing L1TPFCaloProducer::readHcalHGCTCs_ done on top of this.

**Pull request merged**